### PR TITLE
Cart page 06

### LINKED ISSUE #11: feat : "Cart Page"

### PR DESCRIPTION
**Pull request by sadathshariff**
Implemented Cart page
used API for 
1. Add To Cart,
2. Delete
3. Increment and Decrement Quantity

### CODE REVIEW

File: src/components/Card/HorizontalCard/HorizontalCard.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        cartDispatch({ type: "REMOVE_FROM_CART", payload: deletedItem });
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  const updateProductQty = async (id, type) => {
+    try {
+      const updatedCartQty = await updateCartQty(id, type, token);
+      console.log(updatedCartQty);
```
Comment (nishasen) [inline src/components/Card/HorizontalCard/HorizontalCard.jsx]: remove the console.logs

File: src/pages/Cart/Cart.jsx [hunk 2]
```diff
@@ -1,10 +1,76 @@
...
+
+  const totalCostOfProduct = cart.reduce((acc, curr) => {
+    acc = acc + curr.price * curr.qty;
+    return acc;
+  }, 0);
+
+  const totalDiscount = cart.reduce((acc, curr) => {
+    acc = acc + curr.discount * curr.qty;
+    return acc;
+  }, 0);
```
Comment (nishasen) [inline src/pages/Cart/Cart.jsx]: instead of multiple iteration of same array, use reduce in one time and calculate everything.

**Review by nishasen — APPROVED** [review]
Code looks clean and good. I liked how you are calling api and setting the value. A change suggested.

File: src/components/Card/HorizontalCard/HorizontalCard.jsx (hunk 1 above)
Comment (sadathshariff) [reply src/components/Card/HorizontalCard/HorizontalCard.jsx]: yeah will do that Thanks for pointing it out

File: src/pages/Cart/Cart.jsx (hunk 2 above)
Comment (sadathshariff) [reply src/pages/Cart/Cart.jsx]: Will Implement That

File: src/components/Card/HorizontalCard/HorizontalCard.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      if (updatedCartQty) {
+        cartDispatch({ type: "UPDATE_QTY", payload: updatedCartQty });
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
```
Comment (ttahm3d) [inline src/components/Card/HorizontalCard/HorizontalCard.jsx]: These functions can be moved to context and be passed down as values and called here. and can be reused anywhere

File: src/components/Card/ProductCard/ProductCard.jsx [hunk 4]
```diff
@@ -1,20 +1,52 @@
...
+  };
   return (
-    <Link to="/productDetails">
-      <div className={styles.card} onClick={() => setShowProduct(props)}>
-        <img src={img} alt={name} className="card-img" />
-        <div className="card-close-icon"></div>
+    <div className={styles.card} onClick={() => setShowProduct(props)}>
+      <img src={img} alt={name} className="card-img" />
+      <div className="card-close-icon">
+        {isLiked ? <FaHeart color={"red"} /> : <FaRegHeart />}
```
Comment (ttahm3d) [inline src/components/Card/ProductCard/ProductCard.jsx:28]: isLiked is it part of your products or is it limited to wishlist? If isLiked is set to true it shows isliked for all the users, which is not a good approach of doing things

File: src/components/Card/ProductCard/ProductCard.jsx [hunk 5]
```diff
@@ -1,20 +1,52 @@
...
+      </div>
+      <Link to="/productDetails">
         <div className="card-header">
           <h3 className="headline-3">{name}</h3>
           <p className="small-text-2">Rs.{price}</p>
```
Comment (ttahm3d) [inline src/components/Card/ProductCard/ProductCard.jsx:33]: You can use the ruppee symbol using `&#8377;`

File: src/components/Card/ProductCard/ProductCard.jsx [hunk 6]
```diff
@@ -1,20 +1,52 @@
...
+        <Button
+          btnclass={"btn-success"}
+          name={"Go To Cart"}
+          onClick={() => navigate("/cart")}
+        />
+      ) : (
+        <Button
+          btnclass={"btn-primary"}
+          name={"ADD TO CART"}
+          onClick={() => (loggedIn ? addToCart(props) : navigate("../login"))}
```
Comment (ttahm3d) [inline src/components/Card/ProductCard/ProductCard.jsx:47]: Nice bit of work here.

File: src/contexts/Cart/utilis.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import axios from "axios";
+
+export const postToCart = async (product, token) => {
+  try {
+    const res = await axios.post(
+      `/api/user/cart`,
+      { product },
+      {
+        headers: { authorization: token },
+      }
```
Comment (ttahm3d) [inline src/contexts/Cart/utilis.js:11]: create a varaible and use it across the file

**Review by ttahm3d — COMMENTED** [review]
Looks good overall. Minor suggestions in comments

Liked the way you have separated functions in different files

File: src/components/Card/ProductCard/ProductCard.jsx (hunk 5 above)
Comment (sadathshariff) [reply src/components/Card/ProductCard/ProductCard.jsx:33]: Okay Will do that

File: src/components/Card/ProductCard/ProductCard.jsx (hunk 4 above)
Comment (sadathshariff) [reply src/components/Card/ProductCard/ProductCard.jsx:28]: That i will be changing it while doing wishlist part will remove isLiked

File: src/components/Card/HorizontalCard/HorizontalCard.jsx (hunk 3 above)
Comment (sadathshariff) [reply src/components/Card/HorizontalCard/HorizontalCard.jsx]: Right now using it here only so didn't pass in context will add it later

**Pull request merged**